shared/actions/chat.js

```diff
@@ -238,6 +236,20 @@ function * _incomingMessage (action: IncomingMessage): SagaGenerator<any, any> {
         const conversationIDKey = conversationIDToKey(incomingMessage.convID)
         const message = _unboxedToMessage(messageUnboxed, 0, yourName, conversationIDKey)
 
+        // Is this message for the currently selected conversation? If so, mark
+        // it as read ASAP to avoid badging it -- we don't need to badge, the
+        // user's looking at it already.
+        const selectedSelector = (state: TypedState) => state.chat.get('selectedConversation')
+        const selectedConversationIDKey = yield select(selectedSelector)
+        if (message && message.messageID && conversationIDKey === selectedConversationIDKey) {
```

Comment:
Ah, thanks.  We talked about this a bit in frontend.  I don't think it's good enough for badging to wait until the next conversation select to unbadge the selected and focused messages.  So it'll be more involved to fix, by hooking up to app-level changedFocus events.